openupgrade_scripts/scripts/mass_mailing/14.0.2.2/post-migration.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def set_mailing_medium(env):
+    """The field column exists before migration so the new compute behavior won't be
+    triggered. We need to load the values as it's required in the view"""
+    medium_email = env.ref("utm.utm_medium_email").id
+    openupgrade.logged_query(
+        env.cr,
+        f"""
+        UPDATE mailing_mailing
+        SET medium_id = {medium_email}
+        WHERE mailing_type = 'mail'
```

Comment:
Being required, we should fill all of them independently from the mailing_type, so we should provide an alternative. I would say to put to all the medium email, no matter the mailing_type.

Reply:
Well, the other medium_id is `sms`  and that one is provided by the module `mass_mailing_sms` (https://github.com/odoo/odoo/blob/7874c388db9fc476d38390fe5e27d8009031e620/addons/mass_mailing_sms/models/mailing_mailing.py#L55) and there's where another migration script should be made.

Reply:
Uhm, I'm divided here about tolerating a NOT NULL constraint error until all the modules are updated (if they provide the correct migration scripts - you can have extra custom types or the sms one that you mention -), or putting a default and then overwrite it on successive modules.

The first approach allows to detect further problems, but it will only log an ERROR in the migration log, no matter if it's done correctly.

What do you think?

Reply:
Ah, wait, is it only required on the view, but not on the model?

Reply:
That's it. It's only required from the view

Reply:
Ah,OK, then your approach is better.